src/main/java/org/mifos/processor/bulk/schema/SubscriptionDTO.java

```diff
@@ -0,0 +1,118 @@
+package org.mifos.processor.bulk.schema;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.util.UUID;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class SubscriptionDTO {
+
+    private String id;
+
+    private String roomClass;
+
+    private String roomCode;
+
+    private String srcServiceCode;
+
+    private String srcOperationId;
+
+    private String dstServiceCode;
+
+    private String dstOperationId;
+
+    private String delivery;
+
+    private SubscriptionOptionsDTO options;
+
+    public static Builder builder = new Builder();
+
+    public static class Builder {
```

Comment:
good to see this builder here. However, there are some suggestions
1. naming: use the name of the builder as SubscriptionDTOBuilder.
2. naming: use the word 'with' or 'set' for the setter methods. eg:  withRoomClass(..) or withRoomCode(..) I prefer with over set in Builder
3. Use case: If you are not able to find the need of writing multiple constructors to a class, The builder pattern might be an overhead.
here the SubscriptionDTO is used only in one place in the code ( instantiated only in one way), si IMO, a builder pattern is not needed here, unless you foresee multiple use cases for this class in the future with a lot of member variables being optional.

Reply:
This is part of IM BB story and this DTO might be used at multiple places in future thats why I took this LLD decision of creating a builder pattern.

Reply:
Updated the builder class name.